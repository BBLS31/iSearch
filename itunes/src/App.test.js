import React from "react";
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import search from '../src/Components/Function/util'
import SearchBar from "./Components/SearchPage/searchbar";
import renderer from "react-test-renderer";



test("Fetches api data", async () => {
  try {
    const res = await search(); // Calls the api function and awaits the response
    console.log(res); // Logs the data received to the console

    // Checks if the wrapperType property of the response equals "track"
    expect(res.wrapperType).toEqual("track");
  } catch (error) {
    console.error("Error fetching data: ", error);
  }
});





test("Searchbar renders correctly", () => {
  const tree = renderer.create(<SearchBar />).toJSON(); //renders the WeatherApp component into a JSON object using react-test-renderer
  expect(tree).toMatchSnapshot();//compares the rendered output to a snaphot stored in the snapshots directory, or creates a snapshot if it does not exist
});
