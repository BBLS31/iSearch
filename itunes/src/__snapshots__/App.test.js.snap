// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar renders correctly 1`] = `
<div>
  <h1>
    Search
  </h1>
  <div
    className="searchbar"
  >
    <input
      className="search"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
    <select
      className="search-dd"
      onChange={[Function]}
      value="music"
    >
      <option
        value="music"
      >
        Music
      </option>
      <option
        value="movie"
      >
        Movie
      </option>
      <option
        value="podcast"
      >
        Podcast
      </option>
      <option
        value="audiobook"
      >
        Audio Book
      </option>
      <option
        value="shortFilm"
      >
        shortFilm
      </option>
      <option
        value="tvShow"
      >
        Tv Show
      </option>
      <option
        value="ebook"
      >
        Ebook
      </option>
      <option
        value="all"
      >
        All
      </option>
    </select>
    <button
      className="search-btn"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <div
    className="grid-container"
  />
</div>
`;
